utils/getServerSideProps.ts

```diff
@@ -1,32 +1,18 @@
 import { GetServerSidePropsContext, GetServerSidePropsResult } from 'next';
-import { validateRequest } from '~/backend/auth';
+import { lucia, validateRequest } from '~/backend/auth';
+import { logOutUser } from '~/pages/api/auth/logout';
 import { User } from '~/shared/types';
 import { getUserSchema } from '~/shared/zodSchemas';
 
-const EMPTY_USER_OBJECT: User = {
-  email: '',
-  firstName: '',
-  lastName: '',
-  createdAt: new Date(0),
-  updatedAt: new Date(0),
-  uuid: '',
-  role: '',
-};
-
 export async function getServerSideProps(
   context: GetServerSidePropsContext,
 ): Promise<GetServerSidePropsResult<{ user: User }>> {
   const cookieData = await validateRequest(context.req, context.res);
-  if (!cookieData.user || !cookieData.session.isLoggedIn) {
-    if (
-      context.req.url === '/login' ||
-      context.req.url?.includes('/login.json')
-    ) {
-      return {
-        props: { user: JSON.parse(JSON.stringify(EMPTY_USER_OBJECT)) as User },
-      };
-    }
-
+  if (
+    !cookieData.user ||
+    !cookieData.session ||
+    !cookieData.session.isLoggedIn
+  ) {
```

Comment:
Tähän en laittanut `requireLogin`-funktiota, koska sitten pitäisi lähteä muokkaamaan `requireLogin`ia melko paljon. `context.res`in tyyppi on `ServerResponse<IncomingMessage>`, joka ei automaattisesti sisälltyä `.status()`-methodia, joten tiedon palauttamiseen pitäisi tehdä joku tarkistus

Siksi jätin tämän näin